helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -757,9 +762,22 @@ private void handleEvent(ClusterEvent event, BaseControllerDataProvider dataProv
     }
     event.addAttribute(AttributeName.ControllerDataProvider.name(), dataProvider);
 
-    logger.info(String.format("START: Invoking %s controller pipeline for cluster %s event: %s  %s",
-        manager.getClusterName(), dataProvider.getPipelineName(), event.getEventType(),
-        event.getEventId()));
+    String eventSessionId = event.getAttribute(AttributeName.EVENT_SESSION.name());
+    if (eventSessionId != null) {
+      String managerSessionId = manager.getSessionId();
+      if (!eventSessionId.equals(managerSessionId)) {
```

Comment:
if(eventSessionId != null && manager.getSessionId.equals(eventSessionId)) {
  ...
}

Reply:
`String managerSessionId = manager.getSessionId();` is having `managerSessionId` to be used in if check and log. If we do it in the way you suggested, either we don't log this manager session, or we need `manager.getSessionId()` again in log. `getSessionId` also wait for connected. And if during these two getSessionId, the session id changes, we have different session ids.
I prefer to log manager session id so this is why I put a variable here.

Reply:
that is fine, what I meant is we do not two levels of embeded if.. statements in this case.  

if(eventSessionId != null && managerSessionId.equals(eventSessionId))...

Reply:
I got your point. To achieve one level if statement, we could move managerSessionId outside. Then the scope of managerSessionId is bigger.
```
String eventSessionId = event.getAttribute(AttributeName.EVENT_SESSION.name());
String managerSessionId = manager.getSessionId();
if (eventSessionId != null && !eventSessionId.equals(managerSessionId)) {
...
}
```

But after this if statement we don't need managerSessionId. This what I try to do, following `Effective Java Item - 45 : Minimize the scope of local variables`: https://gist.github.com/kpgalligan/4cf93abd88044df71da5

Reply:
Changed.